runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1111,10 +1111,23 @@ Unsafe.getShort.
 */
 
 bool
-TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSymbol, TR::ResolvedMethodSymbol *callerSymbol, TR::TreeTop * callNodeTreeTop, TR::Node * unsafeCall, TR::DataType type, bool isVolatile, bool needNullCheck, bool isOrdered)
+TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSymbol,
+                                              TR::ResolvedMethodSymbol *callerSymbol,
+                                              TR::TreeTop * callNodeTreeTop,
+                                              TR::Node * unsafeCall,
+                                              TR::DataType type,
+                                              bool isVolatile,
+                                              bool needNullCheck,
+                                              bool isOrdered,
+                                              bool isUnaligned)
    {
    if (isVolatile && type == TR::Int64 && comp()->target().is32Bit() && !comp()->cg()->getSupportsInlinedAtomicLongVolatiles())
       return false;
+
+   // In general, Z does not permit unaligned accesses
+   if (isUnaligned && comp()->target().cpu.isZ())
```

Comment:
Would you mind making this a codegen query like `getSupportsInlinedAtomicLongVolatiles()` so that each back end needs to opt in? As it stands a new back end would default to doing this transformation for `get/put*Unaligned` even though the safe default is to prevent the transformation

Similarly in `createUnsafeGetWithOffset()`. But I don't think that code generators should necessarily need to opt in for `get` and `put` separately